### PR TITLE
feat(accounts): add PDA

### PR DESCRIPTION
**Pull request by yihau**
1. generate PDA (derived PDA)
2. sign PDA

### CODE REVIEW

**Comment by yihau** [comment]
PDA examples for #10

File: docs/recipes/accounts.md [hunk 1]
```diff
@@ -38,5 +38,75 @@ the amount of data you intend to store in the account.
   </CodeGroupItem>
 </CodeGroup>
 
+## Program Derived Address
+
+[Prgoram dervied address (PDA)][3] is like a normal address but
```
Comment (Jacob-CG) [inline docs/recipes/accounts.md:43]: This information is great. I am hoping to move all this kind of stuff into the bigger ingredient for PDAs, linking to the reference. Goot to keep for now

Comment (Jacob-CG) [reply docs/recipes/accounts.md:43]: Also a few nits on this:
Should be 
`Program derived address(PDA) is like a normal address with the following differences:`

File: docs/recipes/accounts.md [hunk 2]
```diff
@@ -38,5 +38,75 @@ the amount of data you intend to store in the account.
...
+1. falling off ed25519 curve
+2. using program to sign instead of private key
+
+::: tip
+Although PDA is derived by a program id, it doesn't means the PDA is owned by the same program. (Take an example, you can initialize your PDA as a token account which is an account owned by token program)
+:::
+
+### Generate a PDA
+
+To calculate a PDA, you need
```
Comment (Jacob-CG) [inline docs/recipes/accounts.md:54]: Nit:
`To calculate a PDA, you need the following:`

File: docs/recipes/accounts.md [hunk 3]
```diff
@@ -38,5 +38,75 @@ the amount of data you intend to store in the account.
...
+  <CodeGroupItem title="TS" active>
+
+@[code](@/code/accounts/program-derived-address/derived-a-pda/create-program-address.en.ts)
+
+  </CodeGroupItem>
+</CodeGroup>
+
+2. Find Program Address
+
+`findProgramAddress` will add a extra byte in the end of your seed. It starts from 255 to 0 and return the first off-curve public key. You always get the same result if your pass the same program id and seed.
```
Comment (Jacob-CG) [inline docs/recipes/accounts.md:75]: Nit: `You will always get the same result if you pass the same program id and seed`

File: docs/recipes/accounts.md [hunk 4]
```diff
@@ -38,5 +38,75 @@ the amount of data you intend to store in the account.
...
+@[code](@/code/accounts/program-derived-address/derived-a-pda/find-program-address.en.ts)
+
+  </CodeGroupItem>
+</CodeGroup>
+
+### Sign a PDA
+
+We can use `invoke_signed` to sign a PDA.
+
+#### Program
```
Comment (Jacob-CG) [inline docs/recipes/accounts.md:89]: We probably can just explain the program vs client here, then list the client and program in separate code blocks.

So something like
`PDAs can only be signed for within the program. Below is a program example of signing with a PDA and calling the program with the client` and no other explanation

**Review by Jacob-CG — COMMENTED** [review]
Looks great so far. Just a few changes

File: docs/recipes/accounts.md (hunk 1 above)
Comment (yihau) [reply docs/recipes/accounts.md]: fixed at [here](https://github.com/solana-dev-adv/solana-cookbook/pull/37/commits/055c1d42948c4ad45c407d5986e70889926fc68e)

File: docs/recipes/accounts.md (hunk 2 above)
Comment (yihau) [reply docs/recipes/accounts.md]: fixed at [here](https://github.com/solana-dev-adv/solana-cookbook/pull/37/commits/42d7446110f35c59017ead4f16f161062538a1d2)

File: docs/recipes/accounts.md (hunk 3 above)
Comment (yihau) [reply docs/recipes/accounts.md]: fixed at [here](https://github.com/solana-dev-adv/solana-cookbook/pull/37/commits/6adefc5e7e4d1977d88c3db0c497252d0274f227)

File: docs/recipes/accounts.md (hunk 4 above)
Comment (yihau) [reply docs/recipes/accounts.md:89]: fixed at [here](https://github.com/solana-dev-adv/solana-cookbook/pull/37/commits/814ffc3dbf81f54b507fa45c416cf63121573ed4)

**Review by Jacob-CG — APPROVED** [review]

**Pull request merged**